acme/g25/gpio.go

```diff
@@ -0,0 +1,174 @@
+// Package g25 contains GPIO drivers for the Acme Systems Aria G25
+//
+// The Aria G25 contains up to 60 GPIO pins. This package implements all GPI
+// operations such as getting/setting the value, setting the direction and
+// changing the active low. The package provides a mapping between the
+// pinnumber, the atmel ID and the kernel ID.
+// https://www.acmesystems.it/aria
+package g25
+
+import (
+	"fmt"
+
+	"github.com/advancedclimatesystems/io/gpio"
+)
+
+var w gpio.Watcher
+
+// KernelVersion is the version of the kernel, either 2.6x or 3.1x
+type KernelVersion string
+
+const (
+	// V26 represents the kernel version 2.6x
+	V26 KernelVersion = "2.6x"
+	// V31 represents the kernel version 3.1x
+	V31 KernelVersion = "3.1x"
+)
+
+// NewPin creates a new pin with a kernel ID based on the pin name found
+// here: https://www.acmesystems.it/aria. It assumes the kernel has version 3.1x
+// if this is not the case, use the NewPinV26 instead.
+func NewPin(id string) (gpio.GPIO, error) {
+	return newPin(V31, id)
+}
+
+// NewPinV26 creates a new pin with a kernel ID based on the pin name found
+// here: https://www.acmesystems.it/aria. It assumes the kernel has version 2.6x
+// if this is not the case, use the NewPin instead.
+func NewPinV26(id string) (gpio.GPIO, error) {
+	return newPin(V26, id)
+}
+
```

Comment:
To avoid confusing for package users and to improve portability accross different Linux version I think it's better to create only one `NewPin()`. This method then should check on which Linux version the code is executed.